tools/doc/syncIndexes.js

```diff
@@ -0,0 +1,189 @@
+}
+
+function removeCodeBlocks(text) {
+  var $ = cheerio.load(text);
+  $('pre').remove();
+  $('code').remove();
+  return $.html();
+}
+
+function generateLink(module, link) {
+  return [ 
+    module, '.html#', 
+    module, '_',  link 
+  ].join('');
+}
+
+function generateBreadcrumb(hierarchy) {
+
+  if (hierarchy.length === 0) return false;
+
+  var $ = cheerio.load('<span></span>')
+
+  return hierarchy.map(function(h) {
+    $('span')
+      .addClass('type--left')
+      .addClass('type--' + h.type)
+      .attr('data-href', h.link)
+      .html(h.textValue);
+
+    return $.html();
+  }).join('<span class="sep">&gt;</span>');
+}
+
+function fold(arr, acc, importance, docName, hierarchy) {
+  for (var i in arr) {
+    for (var prop in arr[i]) {
+      var current = arr[i][prop];
+
+      // Don't index function parameters
+      if (typeof current === 'object' && current.textRaw) {
+        if (current.type === 'Array'  || 
+            current.type === 'String' ||
+            current.type === 'Number' ||
+            current.type === 'Boolean'||
+            current.type === 'example') continue;
+
+    if ((current.type === 'var' || 
+         current.type === 'method'
+        ) && hierarchy.length === 0) continue;
+
+        var textValue   = current.textRaw,
+            link        = slugify(textValue),
+            type        = current.type || 'properties',
+            description = (current.desc) 
+              ? removeCodeBlocks(current.desc) 
+              : undefined;
+
+        var currentHierarchy = hierarchy.concat({ 
+          textValue: textValue, 
+          type: type
+        });
+
+        var slugModule = (type === 'module' && hierarchy.length === 0)
+          ? slugify(textValue)
+          : slugify(currentHierarchy[0].textValue);
+
+        var module = moduleAlias[slugModule] || slugModule;
+        
+        var obj = {
+          type: type,
+          textValue: textValue,
+          description: description, 
+          importance: importance
+        };
+
+        obj.module = module;
+        obj.objectID = generateObjectID(link, textValue),
+        obj.isStable = (current.stability !== 0);
+        obj.isModule = (type === 'module'); 
+        obj.link = generateLink(obj.module, link);
+
+        currentHierarchy[currentHierarchy.length - 1].link = obj.link;
+
+        obj.breadcrumb = generateBreadcrumb(hierarchy);
+        obj.hierarchy = hierarchy;
+
+        acc.push(obj);
+
+        fold(current, acc, importance + 1, docName, currentHierarchy);
+      }
+    }
+  }
+
+  return acc;
+}
+
+function indexDummy(res) {
+  console.log(JSON.stringify(res));
+}
+
+function indexChunk(obj) {
+  index.saveObjects(obj);
+}
+
+function readAllJson(err, data) {
+  if (err) {
+    throw err;
+  }
+
+  var object  = JSON.parse(data), 
+      indexes = chunk(fold(object, [], 0, markdownName(object.source), []), 5000, null);
+
+  if (process.argv.length > 1 && process.argv[2] == '--dummy') {
```

Comment:
I would rename `--dummy` to `--debug` and maybe add a usage if `process.argv.length > 2` or `--help` ?